### DISCUSSION TITLE
Problems with Embedded Python

### DISCUSSION THREAD

**Discussion by martyhab00** [InteractiveComputerGraphics/SPlisHSPlasH#310] [Q&A]
I am trying to install SPlisHSPlasH just as specified in the documentation via first cloning the git repo, then running the pip install command. Where in this process can I specify the option for the embedded python, or do I need to build it in a different way? I've found it in the pySPlisHSPlasH CMakeLists file but manually changing it there does not do anything. Thanks for any ideas and help!

**Comment by janbender** [comment]
If you want to use the embedded Python feature, you have to clone the repo and build it using CMake and a C++ compiler. Be sure that you enable the embedded Python option in CMake (this requires the definition of the Python executable).

**Reply by martyhab00** [reply]
Thank you for the help, however I am still not able to get that to work, the embedded python option does not show up and I am not sure what I am doing wrong - see attached file.

![Screenshot 2024-03-18 164041](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/assets/163277849/de474476-3467-4920-9bbe-32caf67e1707)

**Comment by janbender** [comment]
Use "Add entry" and add the field PYTHON_EXECUTABLE as file path. Set the value to your python executable path. Then the option should appear.



Am 18. März 2024 16:41:16 MEZ schrieb martyhab00 ***@***.***>:

**Comment by martyhab00** [comment]
Thanks for the response. Sadly, I am still unable to get this to work. I have documented the procedures I am following both when trying to install it via the python method and via the cmake method that are listed in your tutorial (see annex). I am using VS17, python 3.8 and cmake3.28. I have also tried using VS15 and cmake13 as mentioned in the documentation. This, however, also did not fix the issues I am having. Is there still something I am doing wrong?

[procedure_python.pdf](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/files/14720971/procedure_python.pdf)
[procedure_vs.pdf](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/files/14720972/procedure_vs.pdf)

**Comment by martyhab00** [comment]
Update: 
![Screenshot (68)](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/assets/163277849/88d095a5-ad62-43ac-b321-a35674d8447f)
![Screenshot (69)](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/assets/163277849/5278ccab-50f1-4b0c-86dc-c91aa659f57b)
![Screenshot (70)](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/assets/163277849/d2d4f84b-2fa8-4ab7-b40c-a4ece8bf8af3)
I have now gotten the embedded python working using Python 3.11 and Windows 11, however the error when importing the package I still get the DLL error, see images.

**Comment by martyhab00** [comment]
Since I still cannot get pysplishsplash with embedded python support to run, even though I have been trying a lot of different things like replacing the pybind source code in the extern folder with the current version of it (which made some error messages disappear), trying different git commits and Python versions, I have started wondering:

Is SPlisHSPlasH is even designed to be able to work with pysplishsplash and embedded python support at the same time?

A quick reply to that question would be highly appreciated since that could save a lot of effort trying to get something to run that isn't supposed to work!
